lib/src/debug_logger.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:flutter_reactive_ble/src/model/log_level.dart';
+
+class DebugLogger {
+  DebugLogger(
+    String tag,
+    void Function(Object object) print,
+  )   : _tag = tag,
+        _print = print;
+
+  set logLevel(LogLevel logLevel) => _logLevel = logLevel;
+
+  void log(Object message) {
+    if (_logLevel == LogLevel.verbose) {
+      _print('$_tag: $message');
```

Comment:
I'd even consider leaving the responsibility to add `: ` (a colon followed by a space) to the caller/client (whoever creates the object). A bit less enforced constraints we have no good reasons for.

(e.g. `[BLE SCANNING] message` vs. `BLE SCANNING: message`: we don't care, but now the client have _no_ choice)

Reply:
I think that is a nice addition for the future indeed :)